src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -256,16 +262,30 @@ class WordHighlighter {
 				return;
 			}
 
-			if (this.occurrencesHighlight === 'off') {
+			if (this.occurrencesHighlightEnablement === 'off') {
 				// Early exit if nothing needs to be done!
 				// Leave some form of early exit check here if you wish to continue being a cursor position change listener ;)
 				return;
 			}
 
-			this.runDelayer.trigger(() => { this._onPositionChanged(e); });
+			if (this.occurrencesHighlightDelay !== 250) {
```

Comment:
This doesn't seem right that behavior would change much if the value is 249 vs 250?

Reply:
Played with this a bit more, bc I agree, I initially found this a bit messy. I'm pushing a change to just run this 50ms delayer against all triggers, and I don't think there's any perceivable delay (even with mouse clicks which should happen "instantly"). 

This should keep things cleaner, and have the delay for highlighting tied always to the setting value. 0 still debounces the up/down arrow to prevent highlights flashing around and feels snappy when you release your arrow key.